container executor: nil pointer dereference in getDeploymentSpec when Function podSpec omits terminationGracePeriodSeconds

### Describe the bug

The **container executor** panics with a nil pointer dereference when a `Function`'s `spec.podspec` omits `terminationGracePeriodSeconds` (a `*int64` that is legitimately optional in a core `PodSpec`).

In `pkg/executor/executortype/container/deployment.go`, `getDeploymentSpec` dereferences the pointer **before** the `podSpec == nil` guard and **without** a nil check on the field itself:

```go
gracePeriodSeconds := int64(6 * 60)
if *fn.Spec.PodSpec.TerminationGracePeriodSeconds >= 0 {   // panics if the field is nil
    gracePeriodSeconds = *fn.Spec.PodSpec.TerminationGracePeriodSeconds
}
...
if fn.Spec.PodSpec == nil {   // guard comes too late
    return nil, fmt.Errorf("podSpec is not set for function %s", fn.Name)
}
```

Because the executor never creates the Deployment, the function never specializes and the router returns `specialization_failed` / `connection refused`.

### Stack trace

```
panic: runtime error: invalid memory address or nil pointer dereference
github.com/fission/fission/pkg/executor/executortype/container.(*Container).getDeploymentSpec
    pkg/executor/executortype/container/deployment.go:127
github.com/fission/fission/pkg/executor/executortype/container.(*Container).createOrGetDeployment
    pkg/executor/executortype/container/deployment.go:39
github.com/fission/fission/pkg/executor/executortype/container.(*Container).fnCreate
    pkg/executor/executortype/container/containermgr.go:395
```

### To Reproduce

1. Create a `Function` with `InvokeStrategy.ExecutionStrategy.ExecutorType: container` (prebuilt image / GitOps model).
2. Provide a `spec.podspec` that does **not** set `terminationGracePeriodSeconds` (a normal, valid PodSpec — the field is optional).
3. Trigger the function. The executor reconcile panics; no Deployment/Service-backed pod is ever created.

### Expected behavior

A `Function` whose `podspec` omits the optional `terminationGracePeriodSeconds` should get the executor's default grace period (6 min), not crash the reconcile.

### Workaround

Explicitly set `spec.podspec.terminationGracePeriodSeconds` in the `Function` CR.

### Suggested fix

Nil-check the pointer (and/or move the existing `podSpec == nil` guard above the dereference), defaulting to `6*60` when unset:

```go
if fn.Spec.PodSpec != nil && fn.Spec.PodSpec.TerminationGracePeriodSeconds != nil && *fn.Spec.PodSpec.TerminationGracePeriodSeconds >= 0 {
    gracePeriodSeconds = *fn.Spec.PodSpec.TerminationGracePeriodSeconds
}
```

### Version

- Fission **v1.27.0** (`ghcr.io/fission/fission-bundle:v1.27.0`).
- The same unguarded dereference is present on `main` at the time of writing.

--- a/pkg/executor/executortype/container/deployment.go
+++ b/pkg/executor/executortype/container/deployment.go
@@ -123,8 +123,12 @@ func (cn *Container) getDeploymentSpec(ctx context.Context, fn *fv1.Function, ta
 		replicas = *targetReplicas
 	}
 
+	if fn.Spec.PodSpec == nil {
+		return nil, fmt.Errorf("podSpec is not set for function %s", fn.Name)
+	}
+
 	gracePeriodSeconds := int64(6 * 60)
-	if *fn.Spec.PodSpec.TerminationGracePeriodSeconds >= 0 {
+	if fn.Spec.PodSpec.TerminationGracePeriodSeconds != nil && *fn.Spec.PodSpec.TerminationGracePeriodSeconds >= 0 {
 		gracePeriodSeconds = *fn.Spec.PodSpec.TerminationGracePeriodSeconds
 	}
 
@@ -166,10 +170,6 @@ func (cn *Container) getDeploymentSpec(ctx context.Context, fn *fv1.Function, ta
 	rvCount, err := util.ReferencedResourcesRVSum(ctx, cn.kubernetesClient, fn.Namespace, fn.Spec.Secrets, fn.Spec.ConfigMaps)
 	if err != nil {
 		return nil, err
-	}
-
-	if fn.Spec.PodSpec == nil {
-		return nil, fmt.Errorf("podSpec is not set for function %s", fn.Name)
 	}
 
 	container := &apiv1.Container{

--- a/pkg/executor/executortype/container/deployment_test.go
+++ b/pkg/executor/executortype/container/deployment_test.go
@@ -100,6 +100,30 @@ func TestContainerGetDeploymentSpec(t *testing.T) {
 		assert.Equal(t, int32(2), *deployment.Spec.Replicas)
 	})
 
+	t.Run("nil TerminationGracePeriodSeconds falls back to the default", func(t *testing.T) {
+		cn := newTestContainer()
+		fn := newTestContainerFunction()
+		fn.Spec.PodSpec.TerminationGracePeriodSeconds = nil
+		replicas := int32(1)
+
+		deployment, err := cn.getDeploymentSpec(t.Context(), fn, &replicas, "ctr-fn", "default", nil, nil)
+		require.NoError(t, err)
+
+		pod := deployment.Spec.Template.Spec
+		require.NotNil(t, pod.TerminationGracePeriodSeconds)
+		assert.Equal(t, int64(6*60), *pod.TerminationGracePeriodSeconds)
+	})
+
+	t.Run("nil PodSpec returns an error instead of panicking", func(t *testing.T) {
+		cn := newTestContainer()
+		fn := newTestContainerFunction()
+		fn.Spec.PodSpec = nil
+		replicas := int32(1)
+
+		_, err := cn.getDeploymentSpec(t.Context(), fn, &replicas, "ctr-fn", "default", nil, nil)
+		require.Error(t, err)
+	})
+
 	t.Run("istio disables sidecar injection", func(t *testing.T) {
 		cn := newTestContainer()
 		cn.useIstio = true

